content/docs/hooks-effect.md

```diff
@@ -479,6 +479,27 @@ In the future, the second argument might get added automatically by a build-time
 >
 >We recommend using the [`exhaustive-deps`](https://github.com/facebook/react/issues/14920) rule as part of our [`eslint-plugin-react-hooks`](https://www.npmjs.com/package/eslint-plugin-react-hooks#installation) package. It warns when dependencies are specified incorrectly and suggests a fix.
 
+### Tip: Doing async calls in Effects {#doing-async-calls-in-effects}
+
+Sometimes you need to fetch data for your component when it mounts. Perfect timing to do this would be in `componentDidMount` lifecycle method like this:
+
+```js
+componentDidMount() {
+  this.props.fetchUsers()
+    .then(users => this.setState({ users }))
+}
+```
+
+To do the same with Effects, you can use `useEffect`:
+
+```js
+[users, setUsers] = useState([])
+
+useEffect(() => fetchUsers().then(users => setUsers(users), [])
```

Comment:
@nikolasvk your example code will throw error on unmount. See https://github.com/reactjs/reactjs.org/pull/2125

Reply:
Thanks, I've updated the PR 👍